server/src/main/java/com/cloud/api/dispatch/ParamProcessWorker.java

```diff
@@ -159,6 +159,9 @@ public void processParameters(final BaseCmd cmd, final Map params) {
                 }
                 continue;
             }
+            if (parameterAnnotation.required()){
+                validateNonEmptyString(paramObj, parameterAnnotation.name());
```

Comment:
LGTM - my only worry is if it causes a regression as it create a new behaviour (nit)